static/app/components/organizations/hybridFilter.tsx

```diff
@@ -0,0 +1,397 @@
+                    onClick={() => {
+                      closeOverlay();
+                      commit(stagedValue);
+                    }}
+                  >
+                    {t('Apply')}
+                  </Button>
+                </FooterInnerWrap>
+              )}
+            </FooterWrap>
+          </Fragment>
+        )
+      : null;
+  }, [
+    options,
+    commit,
+    stagedValue,
+    removeStagedChanges,
+    menuFooter,
+    menuFooterMessage,
+    hasStagedChanges,
+    multiple,
+    modifierTipSeen,
+  ]);
+
+  const sectionToggleWasPressed = useRef(false);
+  const onSectionToggle = useCallback(() => (sectionToggleWasPressed.current = true), []);
+
+  const handleChange = useCallback(
+    (selectedOptions: SelectOption<Value>[]) => {
+      const oldValue = stagedValue;
+      const newValue = selectedOptions.map(op => op.value);
+      const oldValueSet = new Set(oldValue);
+      const newValueSet = new Set(newValue);
+
+      // Find out which options were added/removed by comparing the old and new value
+      newValueSet.forEach(val => {
+        if (oldValueSet.has(val)) {
+          newValueSet.delete(val);
+          oldValueSet.delete(val);
+        }
+      });
+      const diff = newValueSet.size > 0 ? [...newValueSet] : [...oldValueSet];
+
+      // A section toggle button was clicked
+      if (diff.length > 1 || sectionToggleWasPressed.current) {
+        sectionToggleWasPressed.current = false;
+        commit(newValue);
+        return;
+      }
+
+      // A modifier key is being pressed --> enter multiple selection mode
+      if (multiple && modifierKeyPressed) {
+        !modifierTipSeen && setModifierTipSeen(true);
+        toggleOption(diff[0]);
+        return;
+      }
+
+      // Only one option was clicked on --> use single, direct selection mode
+      onReplace?.(diff[0]);
+      commit(diff);
+    },
+    [
+      commit,
+      stagedValue,
+      toggleOption,
+      onReplace,
+      multiple,
+      modifierKeyPressed,
+      modifierTipSeen,
+      setModifierTipSeen,
+    ]
+  );
+
+  const handleClear = useCallback(() => {
+    onClear?.();
+    commit([]);
+  }, [onClear, commit]);
+
+  return (
+    <CompactSelect
+      grid
+      multiple
+      clearable={multiple}
+      closeOnSelect={!(multiple && modifierKeyPressed)}
+      options={mappedOptions}
+      value={stagedValue}
+      onChange={handleChange}
+      onClear={handleClear}
+      onSectionToggle={onSectionToggle}
+      onInteractOutside={commitStagedChanges}
+      menuFooter={renderFooter}
+      onKeyDown={onKeyDown}
+      onKeyUp={onKeyUp}
+      {...selectProps}
+    />
+  );
+}
+
+/**
+ * Whether a key that modifies the selection behavior (ctrl/alt/shift) is being pressed.
+ * Based on:
+ * https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/selection/src/utils.ts
+ */
+function isModifierKeyPressed(e: React.KeyboardEvent<HTMLDivElement>) {
```

Comment:
This might be useful to have as a global util

Reply:
https://github.com/getsentry/sentry/pull/47195/commits/dce6ee907109f90701167b7640eefb072e3c065f